src/sentry/tasks/integrations.py

```diff
@@ -154,3 +156,66 @@ def kick_off_status_syncs(project_id, group_id, **kwargs):
                 'external_issue_id': external_issue_id,
             }
         )
+
+
+@instrumented_task(
+    name='sentry.tasks.integrations.kickoff_vsts_subscription_check',
+    queue='integrations',
+    default_retry_delay=60 * 5,  # TODO(lb): not sure what this should be
+    max_retries=5,
+)
+@retry()
+def kickoff_vsts_subscription_check():
+    organization_integrations = OrganizationIntegration.objects.filter(
+        integration__provider='vsts',
+        # integration__status=ObjectStatus.VISIBLE,
+        # status=ObjectStatus.VISIBLE,
+    ).select_related('integration')
+    update_interval = time() - 60 * 60 * 60 * 6
+    for org_integration in organization_integrations:
+        organization_id = org_integration.organization_id
+        integration = org_integration.integration
+        try:
+            subscription = org_integration.integration.metadata['subscription']
+        except KeyError:
+            continue
+
+        try:
+            if subscription['check'] > update_interval:
```

Comment:
Can just add `'check' in subscription and` and cut out the try/except block. Don't really need to use exceptions here.

Reply:
https://docs.python.org/3.6/glossary.html#term-eafp It's easier to ask for forgiveness than permission in python. Different from look before you leap often seen in C/Java that relies more on if-statements rather than try-catch. Because of the way this particular situation is done, I'd say it's on the border of needing to be readable versus efficiency. 

The top answer here has a nice contrasting example.
https://stackoverflow.com/questions/11360858/what-is-the-eafp-principle-in-python


One way I could think to rewrite this with (mostly) LBYL is:
```python
try:
   if 'subscription' not in integration.metadata or integration.metadata['subscription']['check'] > six_hours_from_now:
      continue
except KeyError:
   pass
```
If that's preferable let me know.